firewood/src/merkle/trie_hash.rs

```diff
@@ -26,22 +23,21 @@ impl std::ops::Deref for TrieHash {
 impl Storable for TrieHash {
     fn hydrate<T: CachedStore>(addr: usize, mem: &T) -> Result<Self, ShaleError> {
         let raw = mem
-            .get_view(addr, Self::MSIZE)
+            .get_view(addr, U64_TRIE_HASH_LEN)
             .ok_or(ShaleError::InvalidCacheView {
                 offset: addr,
-                size: Self::MSIZE,
+                size: U64_TRIE_HASH_LEN,
             })?;
-        Ok(Self(
-            raw.as_deref()[..Self::MSIZE as usize].try_into().unwrap(),
-        ))
+
+        Ok(Self(raw.as_deref()[..TRIE_HASH_LEN].try_into().unwrap()))
     }
 
     fn dehydrated_len(&self) -> u64 {
-        Self::MSIZE
+        U64_TRIE_HASH_LEN
     }
 
-    fn dehydrate(&self, to: &mut [u8]) -> Result<(), ShaleError> {
-        Cursor::new(to).write_all(&self.0).map_err(ShaleError::Io)
+    fn dehydrate(&self, mut to: &mut [u8]) -> Result<(), ShaleError> {
+        to.write_all(&self.0).map_err(ShaleError::Io)
```

Comment:
`&mut [u8]` already implements `std::io::Write`. No point in using a `Cursor` if we just throw it away afterwards.